### ISSUE TITLE
Cannot go past login page when accessed via Microsoft Teams's website tab

### ISSUE THREAD

**Issue by tforce7171** [microsoft/azurechat#408]
Hi, I have a working AzureChat and want to access its ui through Microsoft Teams.
I created a website tab which points to my AzureChat endpoint.
It shows a login page(first image), but clicking "Microsoft 365" button to login only redirects to a page below (second image) and won't let me login.
![無題 22](https://github.com/user-attachments/assets/77e1fc4c-27cf-446c-bbb8-044029633c91)
![無題 23](https://github.com/user-attachments/assets/f1cf4f77-6662-45eb-acb9-9bc73d232fff)

And then I read [this article](https://learn.microsoft.com/en-us/microsoftteams/troubleshoot/mobile/websites-with-conditional-access-dont-display) and learned logging in does not work well with MS Teams website tab.

So are there any ways to disable or bipass a login page? I just want to see a chat page on MS Teams, and security or privacy is not really that important at this moment.

**Comment by jeremy-mooney-2024** [comment]
@tforce7171, did you ever find a work around for your post above? I too am seeing the behavior where, on a newly deployed Azure Chat instance, I'm getting the loop of "Sign in with Azure Active Directory" and I sign in successfully, only to be directed back to the same blue button. However the difference is that I am not using Teams. I've tried it in Chrome and Edge (standard and incognito), but no luck.

**Comment by tforce7171** [comment]
No, I have not found a work around. But it seems like Teams website does not store cookies thus causing an issue. So may be looking into cookie settings might be an option.
By the way have you followed [these steps?](https://github.com/microsoft/azurechat/blob/main/docs/5-add-identity.md)

**Comment by jeremy-mooney-2024** [comment]
Yes, I was going for the Azure AD Production setup. Thanks for the tip on the cookies. I may look into that further. But I found a newer Azure AI Chat project that I'm interested in testing and it seems to be more active. https://github.com/Azure-Samples/chat-with-your-data-solution-accelerator

**Comment by tforce7171** [comment]
Thanks for the info. I'll take a look at it.

**Comment by oliverlabs** [comment]
I can confirm that this issue is fixed when User Email field is filled in as per @heguro's comment above.

**Comment by toellbe** [comment]
Can't confirm that it works out of the box in Teams Tab App. Not even with heguro's fix for the auth loop regarding the missing email claim. 

Can anyone share what additional changes are necessary to get a login in a Teams Tab App. Teams Client Side Access Token to NodeJS or Cookie adjustments?

**Comment by toellbe** [comment]
Solved it by implementing teams SSO: add additional Teams Identity Provider to NextAuth, configure cookies for iframe / 3rd party support and using teams js SDK to get client side token from teams.

Redirect authentication ist not supported. Maybe implement entra id popup authentication could also work.

More here: 
https://github.com/nextauthjs/next-auth/discussions/4673
